fly/src/syntax.rs

```diff
@@ -57,19 +125,9 @@ pub enum Quantifier {
     Exists,
 }
 
-/// A binder for a quantifier
-#[derive(PartialEq, Eq, Clone, Debug, Hash, PartialOrd, Ord)]
-pub struct Binder {
-    /// Bound name
-    pub name: String,
-    /// Sort for this binder
-    pub sort: Sort,
-}
-
-/// A Term is a temporal-logical formula (in LTL), which can be interpreted as
-/// being a sequence of values of some sort (often bool for example) under a
-/// given signature and an infinite sequence of states (consisting of values for
-/// all the functions in the signature at each point in time).
+/// A Term is an FO-LTL (first-order linear temporal logic) term or formula. The
+/// temporal operators supported are: Prime, Next, Prev, Until, Since, Always,
+/// Eventually (see [`UOp`] and [`BinOp`]).
```

Comment:
I don't like this change to the documentation. I'd like to have some basic explanation of LTL here, and I don't think it's important to list the operators here (leave that for the language reference).

Reply:
I didn't like the previous explanation, and I'm not sure the basic semantics of LTL should be explained in the documentation of Term. At some point, I think the language manual or other documentation should explain the semantics of FO-LTL. And I did find it helpful (even for myself) to list all the temporal operators here.

What do you think about the following compromise:

```
/// A Term is an FO-LTL (first-order linear temporal logic) term or formula. The
/// temporal operators supported are: Prime, Next, Prev, Until, Since, Always,
/// Eventually (see [`UOp`] and [`BinOp`]).
///
/// FO-LTL is an extension of first-order logic where the semantics is given in
/// terms of infinite sequences of models (over a shared universe). A term is
/// interpreted at a particular point (time) in the sequence, and using
/// temporal operators it can also query the past or future. For example
/// exists x. p(x) & (previous !p(x)) & always r(x) means that there exists
/// some element x for which p now holds, but it didn't hold a moment ago,
/// and from this point onwards r keeps holding for x.
```

Reply:
Sure, that looks good to me.

Reply:
Great. I pushed a slightly edited version of this.